apps/steam_plus/steam_plus.star

```diff
@@ -0,0 +1,143 @@
+"""
+Applet: Steam Plus
+Summary: Show Steam user status
+Description: Shows Steam user avatar, name, status, and currently playing.
+Author: Mike Toscano
+"""
+
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("secret.star", "secret")
+
+STEAM_LOGO_PATH = "https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/7a48ed11-13b2-44e3-a982-8d763d500a3e/db6u02a-edde930a-c504-4284-a0ce-bc88e4ba15b2.png?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7InBhdGgiOiJcL2ZcLzdhNDhlZDExLTEzYjItNDRlMy1hOTgyLThkNzYzZDUwMGEzZVwvZGI2dTAyYS1lZGRlOTMwYS1jNTA0LTQyODQtYTBjZS1iYzg4ZTRiYTE1YjIucG5nIn1dXSwiYXVkIjpbInVybjpzZXJ2aWNlOmZpbGUuZG93bmxvYWQiXX0.bH5kQSIO2Fp1YhXcwTgqQCG-n0HADfWmtPIdqxaF_ZI"
+STATUS = ["Offline", "Online", "Busy", "Away"]
+STATUS_COLOR = ["#59707B", "#0a0", "#F67407", "#FFD100"]
+
+def main(config):
+    api_key = secret.decrypt("AV6+xWcEwaY3vnJsrpZ955musUx4antJ6gbwX3x2owwWA3+5Op9iKbtJOOwUi5IEtAyFUoBZ2m/9udqMTty57El86cAySE44VGeMQY7/7gkzdTxISb1jxZy+bI+TM0sOdoYgA/Fs1kVAnT/JFrG0RESwBcHDaP09o77VlvVv0ujLqnz6x20=")
+
+    STEAM_API_ENDPOINT = "http://api.steampowered.com/ISteamUser/GetPlayerSummaries/v0002/?key=" + api_key + "&steamids="
+    STEAM_GAMES_ENDPOINT = "http://api.steampowered.com/IPlayerService/GetOwnedGames/v0001/?key=" + api_key + "&steamid="
+
+    resp = http.get(STEAM_API_ENDPOINT + config.str("id", ""), ttl_seconds = 180)
+
+    if resp.status_code != 200:
+        fail("Cannot load the specified user")
+
+    players = resp.json()["response"]["players"]
+
+    username = "Cannot find the specified user"
+    avatar = http.get(STEAM_LOGO_PATH).body()
+    currently_playing_logo = http.get(STEAM_LOGO_PATH).body()
```

Comment:
This same URL is retrieved multiple times by the app. Might be good to just get it once and reuse the data where it's needed. Not a huge issue though, as it will be cached on the first request.

Reply:
Good point, resolved in 9e63fddf3c1585392cd163ca6a86d9b6cb909329